nvi-rest/src/main/java/no/sikt/nva/nvi/rest/fetch/Excel.java

```diff
@@ -0,0 +1,104 @@
+package no.sikt.nva.nvi.rest.fetch;
+
+import static nva.commons.core.attempt.Try.attempt;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.lang.reflect.RecordComponent;
+import java.util.List;
+import java.util.function.Consumer;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.usermodel.XSSFCell;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+public class Excel implements AutoCloseable {
+
+    private final Workbook workbook;
+
+    private Excel(Workbook workbook) {
+        this.workbook = workbook;
+    }
+
+    public static <T> Excel fromRecord(List<T> records) {
+        var workbook = new XSSFWorkbook();
+        var className = records.get(0).getClass();
+        if (className.isAnnotationPresent(MyWorkbook.class) && className.isRecord()) {
+            var sheet = workbook.createSheet();
+            var recordComponents = className.getRecordComponents();
+            var headerRow = sheet.createRow(0);
+            for (var colIndex = 0; colIndex < recordComponents.length; colIndex++) {
+                headerRow.createCell(colIndex).setCellValue(getHeaderValue(recordComponents[colIndex]));
+            }
+            for (var rowIndex = 0; rowIndex < records.size(); rowIndex++) {
+                var record = records.get(rowIndex);
+                var row = sheet.createRow(rowIndex + 1);
+                for (var colIndex = 0; colIndex < recordComponents.length; colIndex++) {
+                    var recordComponent = recordComponents[colIndex];
+                    var cell = row.createCell(colIndex);
+                    addValueToCell(recordComponent, record, cell);
+                }
+            }
```

Comment:
I'm torn. The object I'm refering to is a RecordComponent. And to differ from the "normal" understanding of component I think I will stick with recordComponent. Especially since recordComponent is something new with newer java, and not normally associated with reflection.